### ISSUE TITLE
failed to commit (LockFailedException)

### ISSUE THREAD

**Issue by cyberb** [maks/MGit#419]
Big repo: https://github.com/syncthing/syncthing

Almost every operation gives some kind of error.

````
USER_COMMENT=
ANDROID_VERSION=7.0
APP_VERSION_NAME=1.5.4.7
BRAND=motorola
PHONE_MODEL=Moto G (4)
CUSTOM_DATA=
STACK_TRACE=org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of add command
at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:212)
at org.eclipse.jgit.api.CommitCommand.call(CommitCommand.java:203)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.commit(CommitChangesTask.java:97)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.commit(CommitChangesTask.java:56)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.doInBackground(CommitChangesTask.java:44)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.doInBackground(CommitChangesTask.java:20)
at android.os.AsyncTask$2.call(AsyncTask.java:304)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
at java.lang.Thread.run(Thread.java:761)
Caused by: org.eclipse.jgit.errors.LockFailedException: Cannot lock /storage/emulated/0/projects/syncthing-main/.git/index
at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:224)
at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:301)
at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:267)
at org.eclipse.jgit.lib.Repository.lockDirCache(Repository.java:1052)
at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:142)
... 11 more
org.eclipse.jgit.errors.LockFailedException: Cannot lock /storage/emulated/0/projects/syncthing-main/.git/index
at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:224)
at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:301)
at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:267)
at org.eclipse.jgit.lib.Repository.lockDirCache(Repository.java:1052)
at org.eclipse.jgit.api.AddCommand.call(AddCommand.java:142)
at org.eclipse.jgit.api.CommitCommand.call(CommitCommand.java:203)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.commit(CommitChangesTask.java:97)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.commit(CommitChangesTask.java:56)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.doInBackground(CommitChangesTask.java:44)
at me.sheimi.sgit.repo.tasks.repo.CommitChangesTask.doInBackground(CommitChangesTask.java:20)
at android.os.AsyncTask$2.call(AsyncTask.java:304)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
at java.lang.Thread.run(Thread.java:761)
````

**Comment by cyberb** [comment]
This helped, thanks.
This repo was producing crash reports all the time, try cloning and making few commits.
If that works for you without a problem than I probably have some issues with my phone/internal memory.

**Comment by maks** [comment]
Going to track the root cause and hopefully a fix in #503
